OpenRA.Mods.Common/Activities/Rearm.cs

```diff
@@ -20,21 +20,18 @@ namespace OpenRA.Mods.Common.Activities
 	public class Rearm : Activity
 	{
 		readonly AmmoPool[] ammoPools;
-		readonly Dictionary<AmmoPool, int> ammoPoolsReloadTimes;
 
 		public Rearm(Actor self)
 		{
-			ammoPools = self.TraitsImplementing<AmmoPool>().Where(p => !p.Info.SelfReloads).ToArray();
+			ammoPools = self.TraitsImplementing<AmmoPool>().Where(p => !p.SelfReloads).ToArray();
 
 			if (ammoPools == null)
```

Comment:
Wait... `x.Where().ToArray()` can never return `null` ??
Also you have no code left in this method after this, so just get rid of the `if` and `return`.